cli/src/main.rs

```diff
@@ -1,62 +1,128 @@
-//! A simple example of hooking up stdin/stdout to a WebSocket stream.
-//!
-//! This example will connect to a server specified in the argument list and
-//! then forward all data read on stdin to the server, printing out all data
-//! received on stdout.
-//!
-//! Note that this is not currently optimized for performance, especially around
-//! buffer management. Rather it's intended to show an example of working with a
-//! client.
-//!
-//! You can use this example together with the `server` example.
-
-// use std::env;
-
-use futures_util::{future, pin_mut, StreamExt};
-use tokio::io::{AsyncReadExt, AsyncWriteExt};
-use tokio_tungstenite::{connect_async, tungstenite::protocol::Message};
+mod commands;
+mod util;
+
+use futures_util::{future, SinkExt, StreamExt};
+use tokio::io::{AsyncBufReadExt, AsyncWriteExt};
+use tokio_tungstenite::{connect_async, MaybeTlsStream, tungstenite::protocol::Message, WebSocketStream};
 use futures_channel::mpsc;
+use futures_channel::mpsc::{UnboundedReceiver, UnboundedSender};
+use futures_util::stream::{SplitSink, SplitStream};
+use tokio::io;
+use tokio::net::TcpStream;
+use crate::commands::Command;
 
 #[tokio::main]
 async fn main() {
-    let connect_addr = "ws://localhost:3000";
-    let url = url::Url::parse(&connect_addr).unwrap();
+    let (stdin_tx, stdin_rx) = mpsc::unbounded::<Message>();
+    let (write, read) = server_connection().await;
+
+    let stdin_to_ws = tokio::spawn(write_text_to_server(write, stdin_rx));
+    let listen_stream = tokio::spawn(listen_to_server(read));
+
+    let cli_sender = stdin_tx.clone();
```

Comment:
we dont need to clone this unless we want to have another producer. In this case there's a single producer